pkg/extensions/sync/sync_test.go

```diff
@@ -251,6 +251,97 @@ func startDownstreamServer(
 	return dctlr, destBaseURL, destDir, client
 }
 
+func TestORAS(t *testing.T) {
+	Convey("Verify sync on demand for oras objects", t, func() {
+		sctlr, srcBaseURL, _, _, srcClient := startUpstreamServer(t, false, false)
+
+		test.WaitTillServerReady(srcBaseURL)
+
+		defer func() {
+			sctlr.Shutdown()
+		}()
+
+		content := []byte("{\"name\":\"foo\",\"value\":\"bar\"}")
+
+		fileDir := t.TempDir()
+
+		err := os.WriteFile(path.Join(fileDir, "config.json"), content, 0o600)
+		if err != nil {
+			panic(err)
+		}
+
+		content = []byte("hello world")
+
+		err = os.WriteFile(path.Join(fileDir, "artifact.txt"), content, 0o600)
+		if err != nil {
+			panic(err)
+		}
+
+		defer os.Remove(path.Join(fileDir, "config.json"))
+		defer os.Remove(path.Join(fileDir, "artifact.txt"))
+
+		cmd := exec.Command("oras", "version")
+
+		err = cmd.Run()
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Println(fileDir)
+
+		destURL := strings.Join([]string{sctlr.Server.Addr, "/oras-artifact:v2"}, "")
+
+		cmd = exec.Command("oras", "push", destURL, "--manifest-config",
+			"config.json:application/vnd.acme.rocket.config.v1+json", "artifact.txt:text/plain", "-d", "-v")
+		cmd.Dir = fileDir
+
+		// Pushing ORAS artifact to upstream
+		err = cmd.Run()
+		So(err, ShouldBeNil)
+
+		var tlsVerify bool
+
+		regex := ".*"
+
+		syncRegistryConfig := sync.RegistryConfig{
+			Content: []sync.Content{
+				{
+					Prefix: "oras-artifact",
+					Tags: &sync.Tags{
+						Regex: &regex,
+					},
+				},
+			},
+			URLs:      []string{srcBaseURL},
+			TLSVerify: &tlsVerify,
+			CertDir:   "",
+			OnDemand:  true,
+		}
+
+		defaultVal := true
+		syncConfig := &sync.Config{
+			Enable:     &defaultVal,
+			Registries: []sync.RegistryConfig{syncRegistryConfig},
+		}
+
+		dctlr, destBaseURL, _, destClient := startDownstreamServer(t, false, syncConfig)
+
+		test.WaitTillServerReady(destBaseURL)
+
+		defer func() {
+			dctlr.Shutdown()
+		}()
+
+		resp, _ := srcClient.R().Get(srcBaseURL + "/v2/" + "oras-artifact" + "/manifests/v2")
+		So(resp, ShouldNotBeNil)
+		So(resp.StatusCode(), ShouldEqual, 200)
+
+		resp, err = destClient.R().Get(destBaseURL + "/v2/" + "oras-artifact" + "/manifests/v2")
+		So(err, ShouldBeNil)
+		So(resp.StatusCode(), ShouldEqual, 200)
+	})
```

Comment:
after these checks can we also check "oras pull" from destBaseURL (synced one)?

Reply:
Added